src/test/java/redis/clients/jedis/tests/ShardedJedisPoolTest.java

```diff
@@ -103,6 +105,53 @@ public void checkPoolRepairedWhenJedisIsBroken() {
     pool.destroy();
   }
 
+  @Test
+  public void checkRepairedWhenJedisIsBroken() {
+    ShardedJedisPool pool = new ShardedJedisPool(new GenericObjectPoolConfig(), shards);
+    JedisShardInfo shardInfo0 = shards.get(0);
+    JedisShardInfo shardInfo1 = shards.get(1);
+    ShardedJedis jedis1 = pool.getResource();
+    Jedis shard10 = jedis1.getShard(shardInfo0);
+    Jedis shard11 = jedis1.getShard(shardInfo1);
+    breakClient(shard10);
+    assertTrue(shard10.getClient().isBroken());
+    jedis1.close();
+
+    ShardedJedis jedis2 = pool.getResource();
```

Comment:
I'm not sure this is guaranteed by Apache Commons Pool. Maybe need to change the pool config to have size 1 to ensure this.